tree/src/implementations/mpi_helpers.rs

```diff
@@ -0,0 +1,102 @@
+/// Helper Routines for MPI functionality
+use mpi::{
+    datatype::Equivalence,
+    topology::{Communicator, UserCommunicator},
+    traits::*,
+    Count, Rank,
+};
+
+/// Sparse MPI_AllToAllV, i.e. each process only communicates
+/// to a subset of the communicator.
+///
+/// For Example, you may have four processes in a communicator
+/// Communicator = [P0, P1, P2, P3]. Each process communicates
+/// with a subset of the Communicator excluding itself,
+/// ie. P0 -> [P0, P2], P1 -> [P0], P2-> [], P3 -> [P0, P1, P2].
+/// This function expects these packets to be separated in a
+/// Vec<Vec<T>>, their destination ranks, as well as the number of
+/// packets this process expects to receive overall `recv_count`.
+pub fn all_to_allv_sparse<T>(
+    world: &UserCommunicator,
```

Comment:
might it be better to call this `comm` instead of `world` (to avoid confusion if this was called with a subcommunicator)?